### ISSUE TITLE
Create single upload file node

### ISSUE THREAD

**Issue by skylord123** [Skylar-Tech/node-red-contrib-matrix-chat#102]
Instead of having upload image and upload file we should bring them all into one node called upload file that can handle what both of them did. We then need to add options for all the different media types of which there are:

- [ ] m.image
- [ ] m.video
- [ ] m.audio
- [ ] m.file

This node should not post to a room and instead can be chained to a send message node to handle it. This way it can be used to upload room avatar and in the future the user's avatar.

We will have to deprecate the send file and send image nodes. This will be done slowly over the next few releases to make it less annoying. Since now people will need a upload file and send message node chained.

Related docs:
- (Fallback for m.image, m.video, m.audio, and m.file)[https://spec.matrix.org/v1.8/client-server-api/#fallback-for-mimage-mvideo-maudio-and-mfile]
- (m.room.message msgtypes)[https://spec.matrix.org/v1.8/client-server-api/#mroommessage-msgtypes]

**Comment by hillbicks** [comment]
Could you post an example flow for uploading and  "posting" that image? I couldn't find it anywhere in this repo. 

Thanks again for your work on this, I really appreciate it.

**Comment by skylord123** [comment]
Yup sorry about that. The next release has a completely revamped examples directory. I've just been a bit busy lately so haven't been able to get it out.

You can view the example for uploading and sending a file here (Under `Upload file and send to room` -> `Upload file and send to room`)
https://github.com/Skylar-Tech/node-red-contrib-matrix-chat/tree/dev/examples#message-handling

The file upload node can also be used to upload a new user avatar (`User Management` -> `Set user avatar using URL`):
https://github.com/Skylar-Tech/node-red-contrib-matrix-chat/tree/dev/examples#user-management

It can also be used to set the room avatar but looks like I am missing an example for that so I will need to get that added but you essentially just feed the upload image node into a room state events node and have it set the avatar to the uploaded file mxc url.

Hope that gets you going. Let me know if you need any more help.

**Comment by hillbicks** [comment]
That was..... embarrassingly easy. I was pretty sure that I had tested it that way, but apparently not. 

Thanks for your quick answer

**Comment by nickiv** [comment]
The new version of the module is absolutely impressive! Very cool work!

Could you please show an example with `msg.encrypted=true`? I can’t seem to send an image when this flag is enabled. More precisely, the recipient cannot view it and gets an “unable to decrypt” error.

I’m testing this in a direct chat. Your example in the same chat works fine.

**Comment by skylord123** [comment]
@nickiv I will look into this. There may be a bug I missed with the latest release. Let me do some testing and get back to you.

Glad you like the new release :)
